keyboards/nifty_numpad/nifty_numpad.c

```diff
@@ -0,0 +1,247 @@
+/* Copyright 2023 Acliad
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+#ifdef RGB_IDLE_ENABLE
+typedef struct {
+    uint8_t mode;
+    uint8_t speed;
+    HSV     hsv;
+} rgb_matrix_state_t;
+
+typedef struct {
+    rgb_matrix_state_t active_rgb_matrix;
+    rgb_matrix_state_t idle_rgb_matrix;
+} kb_config_t;
+
+static uint32_t idle_timer = 0; 
```

Comment:
instead of using your own timer, you could use `last_input_activity_elapsed()` as this is used/updated by the core based on any activity (matrix scan, encoders, pointing devices, etc)

This would remove some of the necessity for the custom code you have.

Reply:
That's much cleaner, thanks for the heads up! Implemented.